cpp/fetcher/remote_peer.cc

```diff
@@ -62,10 +82,17 @@ void RemotePeer::Impl::DoneGetSTH(
     CHECK_GE(new_sth->tree_size(), 0);
   }
 
-  sth_ = new_sth;
-  if (on_new_sth_) {
-    on_new_sth_(*sth_);
+  if (!sth_ || new_sth->timestamp() > sth_->timestamp()) {
```

Comment:
Nope, we already know it grew (or is > 0) from the CHECKs above, and the code which gets to hear about these new STHs will want to know about same-size-but-newer STHs (so that they can potentially become the mirror's serving STH.)